crates/pgt_workspace/src/workspace/server/change.rs

```diff
@@ -305,10 +314,19 @@ impl Document {
         }
 
         // in any other case, parse the full affected range
-        let changed_content = new_content
-            .as_str()
-            .get(usize::from(full_affected_range.start())..usize::from(full_affected_range.end()))
-            .unwrap();
+        let start_byte = new_content
+            .char_indices()
+            .nth(usize::from(full_affected_range.start()))
+            .map(|(i, _)| i)
+            .unwrap_or(new_content.len());
+
+        let end_byte = new_content
+            .char_indices()
+            .nth(usize::from(full_affected_range.end()))
+            .map(|(i, _)| i)
+            .unwrap_or(new_content.len());
```

Comment:
oh and sanity check: this is inclusive, right?

Reply:
Yes 🙌